public_private_ae2_to_atlas.sh

```diff
@@ -1,60 +1,89 @@
-#!/bin/bash
+#!/usr/bin/env bash
 # @author: rpetry
 # @date:   7 Aug 2013§
 
-# This script serves to reflect in Atlas private/public experiment status switch on the AE2 side.
+shopt -s expand_aliases
+
+# This script serves to reflect in Atlas (gxa and scxa) private/public experiment status switch on the AE2 side.
+# private -> public or public -> private.
+# usage : public_private_ae2_to_atlas.sh <gxa or scxa>
+
 IFS="
 "
 
-if [ $# -ne 5 ]; then
-        echo "Usage: $0 ATLAS_URL AE2_URL ATLAS_ADMIN_UID ATLAS_ADMIN_PASS ERROR_NOTIFICATION_EMAILADDRESS" >&2
+if [ $# -ne 1 ]; then
+        echo "Usage: $0 GXA_OR_SCXA" >&2
         exit 1;
 fi
 
-ATLAS_URL=$1
-AE2_URL=$2
-ATLAS_ADMIN_UID=$3
-ATLAS_ADMIN_PASS=$4
-ERROR_NOTIFICATION_EMAILADDRESS=$5
+## gxa or scxa
+GXA_OR_SCXA=$1
+
+[ ! -z ${ATLAS_URL+x} ] || (echo "Env var ATLAS_URL not defined." && exit 1)
+[ ! -z ${AE2_URL+x} ] || (echo "Env var AE2_URL not defined." && exit 1)
+[ ! -z ${ATLAS_ADMIN_UID+x} ] || (echo "Env var ATLAS_ADMIN_UID not defined." && exit 1)
+[ ! -z ${ATLAS_ADMIN_PASS+x} ] || (echo "Env var ATLAS_ADMIN_PASS not defined." && exit 1)
+[ ! -z ${ERROR_NOTIFICATION_EMAILADDRESS+x} ] || (echo "Env var ERROR_NOTIFICATION_EMAILADDRESS not defined." && exit 1)
+
 
 SUCCESS_HTTP_RESPONSE=200
-process_file="$HOME/tmp/publicprivate_ae2_to_atlas."`eval date +%Y%m%d`
+process_file="$HOME/tmp/${GXA_OR_SCXA}_publicprivate_ae2_to_atlas."`eval date +%Y%m%d`
 rest_results_file=$process_file.rest_results
 all_atlas_experiments_file=$process_file.all_atlas_exps
 all_ae2_experiments_file=$process_file.all_ae2_exps
 exps_not_in_ae2=$process_file.$$.exps_not_in_ae2
 exps_public_in_atlas=$process_file.$$.exps_public_in_atlas
 exps_private_in_atlas=$process_file.$$.exps_private_in_atlas
 
-expsDir=$ATLAS_EXPS
-if [ "$ATLAS_URL" == "ves-hx-77.ebi.ac.uk:8080/gxa" ]; then
-    expsDir=${expsDir}_test
-fi
-
 # Remove any previous $process_file
 rm -rf $process_file
 rm -rf $process_file.log
 touch $process_file.log
 
-# Fetch the list of experiment privacies in Atlas.
- # we will removing PanCancer experiments (E-MTAB-5200 and E-MTAB-5423) as they dont want to be checked, as we want to keep them public
- # in Atlas as they are officially private in ArrayExpress
-fetch_experiment_privacy_from_atlas.pl | grep -v "PROT\|ENAD" | grep -v "E-MTAB-5200\|E-MTAB-5423" > $all_atlas_experiments_file
-if [ $? -ne 0 ]; then
-    echo "Error getting privacy statuses from Atlas to sync with ArrayExpress." >> $process_file.log
-    mailx -s "[gxa/cron] Error getting privacy statuses from Atlas to sync with ArrayExpress." ${ERROR_NOTIFICATION_EMAILADDRESS} < $process_file.log
-    exit 1
+
+if [ $GXA_OR_SCXA == 'gxa' ]; then
+  expsDir=$ATLAS_EXPS
+  if [ "$ATLAS_URL" == "ves-hx-77.ebi.ac.uk:8080/gxa" ]; then
```

Comment:
instead of checking for the machine, which could change in the future, I would suggest that we define the expsDir from an external environment variable. The experiment dir will always be defined externally anyway when setting the environment that we are using (ie. scxa_data_tests or equivalent external source).